core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContext.java

```diff
@@ -731,4 +744,10 @@ public CodecRegistry getCodecRegistry() {
   public ProtocolVersion getProtocolVersion() {
     return getChannelFactory().getProtocolVersion();
   }
+
+  @NonNull
+  @Override
+  public Map<String, String> getStartupOptions() {
+    return startupOptionsBuilderRef.get().build();
```

Comment:
The options are now rebuilt on every call. Did you want to leave the door open in case we ever need an option that would change for every new connection?

Reply:
No, that wasn't the intent. I imagine the options won't change over the course of normal operation.